tobac/utils.py

```diff
@@ -894,3 +894,174 @@ def spectral_filtering(
         return (lambda_mn, transfer_function), filtered_field
     else:
         return filtered_field
+
+
+def compress_all(nc_grids, min_dims=2, comp_level=4):
+    """
+    The purpose of this subroutine is to compress the netcdf variables as they are saved.
+    This does not change the data, but sets netcdf encoding parameters.
+    We allocate a minimum number of dimensions as variables with dimensions
+    under the minimum value do not benefit from tangibly from this encoding.
+
+    Parameters
+    ----------
+    nc_grids : xarray.core.dataset.Dataset
+        Xarray dataset that is intended to be exported as netcdf
+
+    min_dims : integer
+        The minimum number of dimesnions, in integer value, a variable must have in order
+        set the netcdf compression encoding.
+    comp_level : integer
+        The level of compression. Default values is 4.
+
+    Returns
+    -------
+    nc_grids : xarray.core.dataset.Dataset
+        Xarray dataset with netcdf compression encoding for variables with two (2) or more dimensions
+
+    """
+
+    for var in nc_grids:
+        if len(nc_grids[var].dims) >= min_dims:
+            # print("Compressing ", var)
+            nc_grids[var].encoding["zlib"] = True
+            nc_grids[var].encoding["complevel"] = comp_level
+            nc_grids[var].encoding["contiguous"] = False
+    return nc_grids
+
+
+def standardize_track_dataset(TrackedFeatures, Mask, Projection=None):
```

Comment:
I have mixed feelings about this function. On the one hand, I think this will help our users as we complete the (admittedly slow) transition to v2.0. On the other hand, I worry that it commits us to a data structure that we haven't really had a chance to discuss or experiment with. Maybe the way to go is to just label this as experimental and essentially state that we don't commit to keeping this data structure with v2.0? I'm not sure.

Reply:
I was also reflecting a bit on this and I like Sean's idea with labeling it as experimental to make sure we do not commit to this